File: exocps.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace PaymentSystem
{
    public class Payment
    {
        public int StudentId { get; set; }
        public int Amount { get; set; }

        public override string ToString() => $"{StudentId},{Amount}";
    }

    class Program
    {
        private const string FilePath = "payments.txt";

        static void Main()
        {
            if (!File.Exists(FilePath))
                File.Create(FilePath).Close();

            while (true)
            {
                DisplayMenu();
                string choice = Console.ReadLine();

                switch (choice)
                {
                    case "1":
                        MakePayment();
                        break;

                    case "2":
                        Console.Write("Enter Student ID to delete: ");
                        if (int.TryParse(Console.ReadLine(), out int idToDelete))
                            DeletePayment(idToDelete);
                        else
                            Console.WriteLine("Invalid ID!");
                        break;

                    case "3":
                        Console.Write("Enter Student ID to edit: ");
                        if (int.TryParse(Console.ReadLine(), out int idToEdit))
                            EditPayment(idToEdit);
                        else
                            Console.WriteLine("Invalid ID!");
                        break;

                    case "4":
                        Console.Write("Enter Student ID to search: ");
                        if (int.TryParse(Console.ReadLine(), out int idToSearch))
                            SearchPayment(idToSearch);
                        else
                            Console.WriteLine("Invalid ID!");
                        break;

                    case "5":
                        PrintPayments();
                        break;

                    case "6":
                        Console.WriteLine("Exiting...");
                        return;

                    default:
                        Console.WriteLine("Invalid choice!");
                        break;
                }
            }
        }

        static void DisplayMenu()
        {
            Console.WriteLine("*****************************************************************");
            Console.WriteLine("*             Online College Payment System                     *");
            Console.WriteLine("*****************************************************************");
            Console.WriteLine("*                                                               *");
            Console.WriteLine("*        1 - Process a Payment                                  *");
            Console.WriteLine("*        2 - Delete Payment                                     *");
            Console.WriteLine("*        3 - Edit Payment                                       *");
            Console.WriteLine("*        4 - Search Payment                                     *");
            Console.WriteLine("*        5 - Print Payments                                     *");
            Console.WriteLine("*        6 - Exit                                               *");
            Console.WriteLine("*                                                               *");
            Console.WriteLine("*****************************************************************");
            Console.Write("Enter your choice: ");
        }

        static void MakePayment()
        {
            var payment = new Payment();

            Console.Write("Enter Student ID: ");
            string studentIdInput = Console.ReadLine();
            if (!int.TryParse(studentIdInput, out int studentId))
            {
                Console.WriteLine("Invalid ID!");
                return;
            }
            payment.StudentId = studentId;

            Console.Write("Enter Payment Amount: RM");
            string amountInput = Console.ReadLine();
            if (!int.TryParse(amountInput, out int amount))
            {
                Console.WriteLine("Invalid amount!");
                return;
            }
            payment.Amount = amount;

            File.AppendAllText(FilePath, payment + Environment.NewLine);
            Console.WriteLine("\nPayment successfully processed.\n");
        }

        static void DeletePayment(int studentId)
        {
            var lines = File.ReadAllLines(FilePath);
            var newLines = new List<string>();
            bool found = false;

            foreach (var line in lines)
            {
                var parts = line.Split(',');
                if (parts.Length == 2 && int.Parse(parts[0]) != studentId)
                    newLines.Add(line);
                else
                    found = true;
            }

            File.WriteAllLines(FilePath, newLines);
            Console.WriteLine(found ? "Payment deleted successfully!" : "Payment not found!");
        }

        static void EditPayment(int studentId)
        {
            var lines = File.ReadAllLines(FilePath);
            bool found = false;

            for (int i = 0; i < lines.Length; i++)
            {
                var parts = lines[i].Split(',');
                if (parts.Length == 2 && int.Parse(parts[0]) == studentId)
                {
                    Console.Write($"Enter new amount for {studentId}: RM");
                    if (int.TryParse(Console.ReadLine(), out int newAmount))
                    {
                        lines[i] = $"{studentId},{newAmount}";
                        found = true;
                    }
                    else
                    {
                        Console.WriteLine("Invalid amount!");
                    }
                }
            }

            File.WriteAllLines(FilePath, lines);
            Console.WriteLine(found ? "Payment updated successfully!" : "Payment not found!");
        }

        static void SearchPayment(int studentId)
        {
            var lines = File.ReadAllLines(FilePath);
            bool found = false;

            foreach (var line in lines)
            {
                var parts = line.Split(',');
                if (parts.Length == 2 && int.Parse(parts[0]) == studentId)
                {
                    Console.WriteLine(line);
                    found = true;
                    break;
                }
            }

            if (!found)
                Console.WriteLine("Payment not found!");
        }

        static void PrintPayments()
        {
            var lines = File.ReadAllLines(FilePath);
            int totalAmount = 0;
            int numPayments = 0;

            Console.WriteLine("\nAll Payments:");
            foreach (var line in lines)
            {
                Console.WriteLine(line);
                var parts = line.Split(',');
                if (parts.Length == 2)
                {
                    totalAmount += int.Parse(parts[1]);
                    numPayments++;
                }
            }

            double avgPayment = numPayments > 0 ? (double)totalAmount / numPayments : 0;
            var now = DateTime.Now;

            Console.WriteLine("\n*****************************************************************");
            Console.WriteLine("*               Payment Overview                                *");
            Console.WriteLine("*****************************************************************");
            Console.WriteLine("*                                                               *");
            Console.WriteLine($"*        Total amount paid: RM{totalAmount}                    *");
            Console.WriteLine($"*        Number of payments: {numPayments}                     *");
            Console.WriteLine($"*        Average payment: RM{avgPayment:F2}                    *");
            Console.WriteLine("*                                                               *");
            Console.WriteLine($"*        Time/Date: {now:hh:mm:ss tt MM/dd/yyyy}               *");
            Console.WriteLine("*****************************************************************");
            Console.WriteLine("*        THANK YOU FOR USING THE SERVICE                        *");
            Console.WriteLine("*****************************************************************\n");
        }
    }
}

File: exocps.py
import os
from datetime import datetime

class Payment:
    def __init__(self, student_id=0, amount=0):
        self.student_id = student_id
        self.amount = amount

def display_menu():
    print("*****************************************************************")
    print("*                                                               *")
    print("*             Online College Payment System                     *")
    print("*                                                               *")
    print("*****************************************************************")
    print("*                                                               *")
    print("*        1 - Process a Payment                           *")
    print("*        2 - Delete Payment                              *")
    print("*        3 - Edit Payment                                *")
    print("*        4 - Search Payment                              *")
    print("*        5 - Print Payments                              *")
    print("*        6 - Exit                                        *")
    print("*                                                               *")
    print("*****************************************************************\n")
    print("Enter your choice: ", end="")

def make_payment():
    payment = Payment()
    
    try:
        payment.student_id = int(input("Enter Student ID: "))
        payment.amount = int(input("Enter Payment Amount: RM"))
    except ValueError:
        print("Invalid input. Please enter numbers only.")
        return

    with open("payments.txt", "a") as file:
        file.write(f"{payment.student_id},{payment.amount}\n")
    
    print("\nPayment successfully processed.\n")

def delete_payment(student_id_to_delete):
    found = False
    temp_lines = []
    
    try:
        with open("payments.txt", "r") as file:
            lines = file.readlines()
        
        for line in lines:
            student_id = int(line.split(',')[0])
            if student_id != student_id_to_delete:
                temp_lines.append(line)
            else:
                found = True
        
        with open("payments.txt", "w") as file:
            file.writelines(temp_lines)
            
    except FileNotFoundError:
        print("No payments file found!")
        return
    
    if found:
        print("Payment deleted successfully!")
    else:
        print("Payment not found!")

def edit_payment(student_id_to_edit):
    found = False
    temp_lines = []
    
    try:
        with open("payments.txt", "r") as file:
            lines = file.readlines()
        
        for line in lines:
            parts = line.split(',')
            student_id = int(parts[0])
            
            if student_id == student_id_to_edit:
                found = True
                try:
                    new_amount = int(input(f"Enter New Amount for {student_id}: RM"))
                    temp_lines.append(f"{student_id},{new_amount}\n")
                except ValueError:
                    print("Invalid amount. Keeping original payment.")
                    temp_lines.append(line)
            else:
                temp_lines.append(line)
        
        with open("payments.txt", "w") as file:
            file.writelines(temp_lines)
            
    except FileNotFoundError:
        print("No payments file found!")
        return
    
    if found:
        print("Payment information successfully updated!")
    else:
        print("Payment not found!")

def search_payment(student_id_to_search):
    found = False
    
    try:
        with open("payments.txt", "r") as file:
            for line in file:
                parts = line.strip().split(',')
                student_id = int(parts[0])
                
                if student_id == student_id_to_search:
                    found = True
                    print(line.strip())
                    break
    except FileNotFoundError:
        print("No payments file found!")
        return
    
    if not found:
        print("Payment not found!")

def print_payments():
    num_payments = 0
    total_amount = 0
    
    try:
        with open("payments.txt", "r") as file:
            lines = file.readlines()
            
            if not lines:
                print("No payments found!")
                return
                
            print("\nAll Payments:")
            for line in lines:
                print(line.strip())
                parts = line.strip().split(',')
                amount = int(parts[1])
                num_payments += 1
                total_amount += amount
    except FileNotFoundError:
        print("No payments file found!")
        return
    
    average_payment = total_amount / num_payments if num_payments > 0 else 0
    now = datetime.now()
    
    print("\n*****************************************************************")
    print("*               Payment Overview                                *")
    print("*****************************************************************")
    print("*                                                               *")
    print(f"*        Total amount paid: RM{total_amount}                    *")
    print(f"*        Number of payments: {num_payments}                    *")
    print(f"*        Average payment: RM{average_payment:.2f}              *")
    print("*                                                               *")
    print(f"*        Time/Date: {now.strftime('%I:%M:%S %p %m/%d/%Y')}     *")
    print("*****************************************************************")
    print("*        THANK YOU FOR USING THE SERVICE                        *")
    print("*****************************************************************\n")

def main():
    # Create empty file if it doesn't exist
    if not os.path.exists("payments.txt"):
        open("payments.txt", "w").close()
    
    while True:
        display_menu()
        menu_choice = input().strip()
        
        if menu_choice == '1':
            make_payment()
        elif menu_choice == '2':
            try:
                student_id = int(input("Enter Student ID to delete: "))
                delete_payment(student_id)
            except ValueError:
                print("Invalid Student ID. Please enter a number.")
        elif menu_choice == '3':
            try:
                student_id = int(input("Enter Student ID to edit: "))
                edit_payment(student_id)
            except ValueError:
                print("Invalid Student ID. Please enter a number.")
        elif menu_choice == '4':
            try:
                student_id = int(input("Enter Student ID to search: "))
                search_payment(student_id)
            except ValueError:
                print("Invalid Student ID. Please enter a number.")
        elif menu_choice == '5':
            print_payments()
        elif menu_choice == '6':
            print("Exiting program...")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()
